app/services/cached_appeal_service.rb

```diff
@@ -272,4 +277,69 @@ def veteran_names_for_correspondent_ids(correspondent_ids)
       [corr.stafkey, "#{corr.snamel.split(' ').last}, #{corr.snamef}"]
     end.to_h
   end
+
+  def appeal_ids_with_schedule_hearing_tasks
+    @appeal_ids_with_schedule_hearing_tasks ||= []
+  end
+
+  # get all ids of appeals which have open ScheduleHearingTasks
+  def filter_appeals_by_schedule_hearing_tasks(appeal_ids, appeal_type)
+    appeal_ids_with_schedule_hearing_tasks << Task.open
+      .where(appeal_id: appeal_ids, type: ScheduleHearingTask.name, appeal_type: appeal_type)
+      .pluck(:appeal_id)
+      .uniq
+    appeal_ids_with_schedule_hearing_tasks.flatten!
+  end
+
+  def vacols_id_to_appeal_id_mapping
+    @vacols_id_to_appeal_id_mapping ||= {}
+  end
+
+  # vacols_id => appeal_id mapping
+  def map_vacols_id_to_appeal_id(legacy_appeals)
```

Comment:
Rename method to reflect that it is change this service's state
```suggestion
  def append_vacols_id_appeal_id_mapping(legacy_appeals)
```